app/models/hackathon/subscription/status.rb

```diff
@@ -9,6 +9,22 @@ module Hackathon::Subscription::Status
     scope :active_for, ->(user) { active.where(subscriber: user) }
   end
 
+  # Unsubscribe URLs must be valid for at least 30 days.
+  # https://www.ftc.gov/business-guidance/resources/can-spam-act-compliance-guide-business
+  UNSUBSCRIBE_EXPIRATION = 2.months
+
+  class_methods do
+    def manage_subscriptions_url_for(user)
+      user_signature = user.signed_id purpose: :manage_subscriptions, expires_in: UNSUBSCRIBE_EXPIRATION
+      Rails.application.routes.url_helpers.user_subscriptions_url(user_signature)
+    end
+
+    def unsubscribe_all_url_for(user)
+      user_signature = user.signed_id purpose: :manage_subscriptions, expires_in: UNSUBSCRIBE_EXPIRATION
+      Rails.application.routes.url_helpers.unsubscribe_all_user_subscriptions_url(user_signature)
+    end
+  end
```

Comment:
I don't love having this logic in the model. Could we just use the constant? Also, do we have to have a separate page to unsubscribe completely? Pretty sure it's legal to have one page for managing subscriptions, and then we could just have the button there and/or an option to select all subscriptions.

Reply:
The separate unsubscribe all page seems to be common in emails. I'm not sure if there is some law requiring it, but I decided to follow that design since it appears to be the most popular way to implement unsubscribe links.

> Could we just use the constant?

Could you explain more?

Reply:
As in move the URL functionality out of the model.